File: ruby/sort/selection_sort.rb
def selection_sort(array)
  array.each_index do |i|
    min_idx = i
    ((i + 1)...array.size).each do |j|
      min_idx = j if array[j] < array[min_idx]
    end
    array[i], array[min_idx] = array[min_idx], array[i] unless min_idx == i
  end
  array
end

File: rust/sort/selection_sort.rs
pub fn selection_sort<T: Ord>(array: &mut [T]) {
  for i in 0..array.len() {
    let mut min_idx = i;
    for j in (i + 1)..array.len() {
      if array[j] < array[min_idx] {
        min_idx = j;
      }
    }
    array.swap(i, min_idx);
  }
}